app/models/piece.rb

```diff
@@ -84,13 +84,53 @@ def horizontal_obstruction?(destination_x, _destination_y)
   def move_to!(destination_x, destination_y)
     destination_piece = Piece.find_by(x: destination_x, y: destination_y)
 
-    if destination_piece.present?
-      if destination_piece.color != color
-        destination_piece.destroy
-      else
-        return
+    if valid_move?(destination_x, destination_y)
+      if destination_piece.present?
+        if destination_piece.color != color
+          destination_piece.destroy
+          return true
+        else
+          return false
+        end
       end
+      update(x: destination_x, y: destination_y)
+      return true
     end
-    update(x: destination_x, y: destination_y)
   end
-end
+
+  def move!(destination_x, destination_y, type)
```

Comment:
once you have the `type` set in your test. This method could be 1 line :smile: :

```
move_to!(destination_x, destination_y) if valid_move?(destination_x, destination_y)
```